arbos/storage/storage.go

```diff
@@ -266,6 +289,24 @@ func (store *Storage) Keccak(data ...[]byte) ([]byte, error) {
 	return crypto.Keccak256(data...), nil
 }
 
+// note: returned slice is not thread-safe
+func (store *Storage) cachedKeccak(data ...[]byte) []byte {
```

Comment:
Besides being cached, it doesn't behave exactly the same as `Keccak` or `KeccakHash` methods define on storage (which also seem to burn cost). Please add documentation how they differ and when should one be used.